tests/std/tests/P0226R1_math_special_functions/test.cpp

```diff
@@ -0,0 +1,499 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <cmath>
+#include <limits>
+
+using namespace std;
+
+template <typename Float>
+void assert_close(const Float f, const Float g) {
+    auto eps = 0.000003 * abs(f);
+    if (eps == Float{0}) {
+        eps = numeric_limits<Float>::min();
+    }
+    assert(abs(f - g) < eps);
+}
+
+constexpr float nanf_v       = numeric_limits<float>::quiet_NaN();
+constexpr double nand_v      = numeric_limits<double>::quiet_NaN();
+constexpr long double nanl_v = numeric_limits<long double>::quiet_NaN();
+
+void test_assoc_laguerre() {
+    {
+        assert(isnan(assoc_laguerre(1, 10, nanf_v)));
+        assert(isnan(assoc_laguerre(1, 10, nand_v)));
+        assert(isnan(assoc_laguerre(1, 10, nanl_v)));
+        assert(isnan(assoc_laguerref(1, 10, nanf_v)));
+        assert(isnan(assoc_laguerrel(1, 10, nanl_v)));
+    }
+    {
+        assert_close(assoc_laguerre(1, 10, 0.5), 10.5);
+        assert_close(assoc_laguerre(1, 10, 0.5f), 10.5f);
+        assert_close(assoc_laguerre(1, 10, 0.5L), 10.5L);
+        assert_close(assoc_laguerref(1, 10, 0.5f), 10.5f);
+        assert_close(assoc_laguerrel(1, 10, 0.5L), 10.5L);
+    }
+
+    {
+        assert_close(assoc_laguerre(2, 10, 0.5), 60.125);
+        assert_close(assoc_laguerre(2, 10, 0.5f), 60.125f);
+        assert_close(assoc_laguerre(2, 10, 0.5L), 60.125L);
+        assert_close(assoc_laguerref(2, 10, 0.5f), 60.125f);
+        assert_close(assoc_laguerrel(2, 10, 0.5L), 60.125L);
+    }
+}
+
+void test_assoc_legendre() {
+    {
+        assert_close(assoc_legendre(2, 0, 0.5), -0.125);

```

Comment:
For example, here we should get this answer <https://www.wolframalpha.com/input?i=LegendreP%5B2%2C+0%2C+0.5%5D>